Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,35 @@ def installed?
       !versions.empty?
     end
 
+    def outdated?(greedy = false)
+      !outdated_versions(greedy).empty?
+    end
+
+    def outdated_versions(greedy = false)
+      # special case: tap version is not available
+      return [] if version.nil?
+
+      # if --greedy is not present in the command line then skip when detection is not safe
+      unless greedy
```

Comment:
This should be:

```ruby
      if greedy
        return versions if version.latest?
      else
        return [] if auto_updates
      end
```

Otherwise, when I have the `:latest` version installed, but there is an updated cask with version `1.0.0`, it will not show up as outdated.